x-pack/plugins/security_solution/public/management/pages/trusted_apps/store/selectors.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Immutable } from '../../../../../common/endpoint/types';
+import { TrustedAppsPageState } from '../state/trusted_apps_page_state';
+import { pageInfosEqual } from '../state/items_page';
+import { isStaleBinding } from '../state/async_data_binding';
+
+export const needsRefreshOfListData = (state: Immutable<TrustedAppsPageState>) => {
+  const currentPageInfo = state.list.currentPageInfo;
+  const currentPage = state.list.currentPage;
+
+  return (
+    state.active &&
+    isStaleBinding(currentPage, (data) => pageInfosEqual(currentPageInfo, data.pageInfo))
```

Comment:
it seems like this is ultimately based on `pageInfosEqual` which checks that we have the same amount of entries and on the same index?  We could very well have different data with these conditions.

My understanding is that this checks if we should refresh the data, so we're checking if the lates set of fetch data is any different than the current data.  I'm not sure the check would work in all cases, such as when we remove and then add a new item in between checks, we'd have the same number of entries, but they would be different lists.  Let me know if I'm missing something

Reply:
The data stored in the state of redux is immutable, so whatever server returned as a response can not be changed without having an explicit action that would support that (TypeScript in combination with Redux will prevent that). I would strongly discourage modifying data structured sent from the server, I'd rather suggest we refresh data from server in the case we modified something (like added trusted app), because there is no reliable way to predict where is the item supposed to end up.

In case you mean that when adding or deleting items the current location (page and page size) did not change and this condition would not trigger refresh you are right. It's a bit out of scope of this story but there are couple of strategies to address that:
   * My preferable: In addition to page information we store timestamp (could be time when data was requested to refresh), this timestamp would be considered in the data freshness check, when we delete or add items we update this timestamp on success. Instead of timestamp it could also be a boolean flag, but timestamp has nice property that it can be used also for other purposes like periodic refresh, etc.
   * When addition or deletion succeeds we could also just explicitly invoke refreshList in the middleware. This is a little bit less deterministic and I would personally try to see us try the first option.

Still I would suggest we take this discussion outside of the scope of this story and for example pick it up right away with the deletion story. What do you think?